### PR TITLE
Add Prometheus data model

### PR DESCRIPTION
**Pull request by harrytucker**
This PR adds a module with a structure representing a Prometheus alert. This doesn't cover deployment, but this serialises as we expect it to.

### CODE REVIEW

File: src/prometheus/alert.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct Alerts {
+    pub groups: Vec<AlertGroup>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AlertGroup {
```
Comment (cyclingwithelephants) [inline src/prometheus/alert.rs:11]: What's the intention behind AlertGroup?

File: src/prometheus/alert.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AlertRules {
+    pub alert: String,
+    pub expr: String,
+    #[serde(rename = "for")]
+    pub for_: String,
+    pub labels: HashMap<String, String>,
+    pub annotations: HashMap<String, String>,
```
Comment (cyclingwithelephants) [inline src/prometheus/alert.rs:23]: Given that we're not exposing this publically, I wonder if it's worth making annotations a struct with summary and description fields?

File: src/prometheus/alert.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AlertGroup {
+    pub name: String,
+    pub rules: Vec<AlertRules>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AlertRules {
+    pub alert: String,
```
Comment (cyclingwithelephants) [inline src/prometheus/alert.rs:18]: what's the alert field for?

File: src/prometheus/alert.rs (hunk 2 above)
Comment (harrytucker) [reply src/prometheus/alert.rs:23]: I implemented this module intended to be the final thing we produce and serialise for Prometheus (in our MVP, `rule-watcher` ConfigMaps) to ingest. So the whole structure put together maps exactly to a Prometheus alert rule: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/

File: src/prometheus/alert.rs (hunk 3 above)
Comment (harrytucker) [reply src/prometheus/alert.rs:18]: Following the Prometheus structure here from their docs for the final output format, `alert` is the name of a single alert that is part of a group of alerts.

**Pull request merged**